tests/Opw.PineBlog.Core.Tests/Feeds/GetSyndicationFeedQueryTests.cs

```diff
@@ -119,21 +131,20 @@ public async Task Handler_Should_ReturnFeedModel_WithCorrectItemId()
             guid.Should().Be("http://www.example.com/posts/post-title-4");
         }
 
-        private void SeedDatabase()
+        private async Task SeedDatabase()
         {
-            var context = ServiceProvider.GetRequiredService<IBlogEntityDbContext>();
+            var repo = ServiceProvider.GetRequiredService<IRepository>();
+            CancellationToken cancelToken = CancellationToken.None;
 
             var author = new Author { UserName = "user@example.com", DisplayName = "Author 1" };
-            context.Authors.Add(author);
```

Comment:
When implementing a unit of work / repository pattern. We can end up with something like this:
``` csharp
uow.Authors.Add(author);
```
Where `uow` is the unit of work, `Authors` is the repository, and `Add` is the add method.
``` csharp
interface IUnitOfWork
{
  IAuthorRepository Authors { get; }
  ...
  Result<int> SaveChanges();
  Task<Result<int>> SaveChangesAsync(CancellationToken cancellationToken);
}

interface IAuthorRepository
{
  Result<TEntity> Add([NotNullAttribute] TEntity entity);
  ...
}
```

Reply:
I'm unsure how this can be implemented as you describe due to how Mongo's transactions are implemented: https://docs.mongodb.com/manual/core/transactions/ Notably, individual actions are done within a transaction callback.

Reply:
I've implemented a unit of work and did a test for MongoDb, to see if it would work nicely. It was not too much work. With this implementation we can abstract EF and MongoDb fully, none of the other project should need any dependencies on any data acces technology, so that is nice! I did not implement the MongoDb part completely and also did not add tests for that.

You can have a look at this PR for my implementation: https://github.com/ofpinewood/pineblog/pull/91